test/functional/wallet_conflicts.py

```diff
@@ -123,5 +130,165 @@ def run_test(self):
         assert_equal(former_conflicted["confirmations"], 1)
         assert_equal(former_conflicted["blockheight"], 217)
 
+    def test_mempool_conflict(self):
+        self.nodes[0].createwallet("alice")
+        alice = self.nodes[0].get_wallet_rpc("alice")
+
+        bob = self.nodes[1]
+
+        self.nodes[2].send(outputs=[{alice.getnewaddress() : 25} for _ in range(3)])
+        self.generate(self.nodes[2], 1)
+
+        self.log.info("Test a scenario where a transaction has a mempool conflict")
+
+        unspents = alice.listunspent()
+        assert_equal(len(unspents), 3)
```

Comment:
71059cf4267d80d2934b85a7046fc7b64900378b

There's not anything preventing us from checking the exact contents of `listunspent` right? Here and in the rest of the test.

Reply:
Good point, the tests now check the exact contents.